### ISSUE TITLE
`<optional>`: Mystery LWG issue citations

### ISSUE THREAD

**Issue by StephanTLavavej** [microsoft/STL#1770]
https://github.com/microsoft/STL/blob/dcba13b91e9f39c16319d41ab62b2b5fde2c0be5/stl/inc/optional#L417-L421
https://github.com/microsoft/STL/blob/dcba13b91e9f39c16319d41ab62b2b5fde2c0be5/stl/inc/optional#L431-L435

* Was an LWG issue filed for this?
  + If not, we should file it.
  + If so, we should cite the number.
  + If it's already been voted into the WP, we should drop "as modified by" entirely.
* We should cite a WP for `[optional.observe]`. (Paragraphs can be renumbered over time.)

**Comment by CaseyCarter** [comment]
[The LWG Index by Section](http://cplusplus.github.io/LWG/lwg-index.html), lists exactly three issues for [[optional.observe]](http://eel.is/c++draft/optional.observe):
* LWG-2740 `constexpr optional<T>::operator->` (Applied in C++17)
* LWG-2829 LWG 2740 leaves behind vacuous words (Open)
* LWG-3424 `optional::value_or` should never return a cv-qualified type (Open)

The PR for LWG-3424 changes `optional<T>::value_or` to return `remove_cv_t<T>` instead of `T`, so it is clearly the issue intended. This change must have happened before we started using the `// per LWG issue unnumbered as of YYYY-MM-DD` convention.

Both of these should be updated to `N4885 [optional.observe] ... modified by LWG-3424".

**Comment by Athira2199** [comment]
Hi @CaseyCarter I would like to contribute to this issue. Could you guide as to exactly how I can contribute. Quiet new to such contributions.

**Comment by StephanTLavavej** [comment]
@Athira2199 You'll need to create a pull request (PR), which might seem complicated if you've never done it before - but you should be able to find many articles/video tutorials to walk you through the process. The actual source code edits you'll need to make will be 4 changes to 2 lines in 1 file. Briefly:

* Fork the microsoft/STL repo, creating Athira2199/STL.
  + You need to create a fork so you can push changes to your fork, then ask microsoft/STL to "pull" those changes.
  + Everyone who contributes to the STL creates such a fork, e.g. https://github.com/StephanTLavavej/STL and https://github.com/CaseyCarter/STL . (But don't fork our forks! Fork microsoft/STL itself.)
* Clone your repo Athira2199/STL to your local drive.
* On your local machine, open the file `stl/inc/optional` that was quoted in my issue submission above.
* Find the lines mentioning:
  + `"([optional.observe]/18 as modified by LWG-XXXX)."`
  + `"([optional.observe]/20 as modified by LWG-XXXX)."`
* As @CaseyCarter mentioned, the edits that you need to make are:
  + Add `N4885`, followed by a space, right before `[optional.observe]`. (This is the document number for the current version of the C++ Working Paper.)
  + `LWG-XXXX` was a placeholder for a Library Working Group issue, which has been created and numbered. This should be replaced with `LWG-3424`.
  + Make these changes to both lines: thus 4 total edits across 2 lines.
* Save the file. `git add` and `git commit` it, writing a short message. Your commit message can be `<optional>: Fix mystery LWG issue citations` or something along those lines.
* For a significant code change, you should build and test your changes. However, for this change specifically, modifying `static_assert` messages, you can get away with skipping a build/test. (It is still possible to break our build - e.g. leaving trailing whitespace in the file - but don't worry about it.)
* Push this commit to your fork.
  + Afterwards, you should be able to view this commit on GitHub on the web, when looking at your fork.
* Go back to the microsoft/STL repo, click on the Pull Requests tab, click on the green "New pull request" button, click "compare across forks", on the **right side** ("head repository", the name is really confusing) select your fork Athira2199/STL, and for the "compare:" branch select the branch you pushed to (by default it will be your `main` branch). Leave the left side set to "base repository: microsoft/STL" (again confusing terminology - this is the intended destination for your changes) and "base: main" (since you want our main branch to pull your changes). There should be a preview of your changes below; inspect it to make sure that the edits you want are there, and there are no stray edits. Click "Create pull request".
* Give the PR a title - again, `<optional>: Fix mystery LWG issue citations` will be fine - and in the body, say "Fixes #1770." (this issue!) which will tell GitHub that your PR will resolve this issue. Click the Preview tab to double-check. If it all looks good, click "Create pull request" and we'll take it from there (reviewing and eventually merging your PR).

It seems like a ton of steps, and it is, but the good news is that by learning this process with a relatively simple source file change, you'll be able to submit pull requests to any other repo on GitHub because the same process is used everywhere (with variations around branches, review process, etc.).

**Comment by Athira2199** [comment]
@StephanTLavavej Thanks a lot for this detailed description. I highly appreciate it.🙂
